service/register.go

```diff
@@ -24,8 +24,7 @@ const (
 func Register(service string, host *Host, stop chan struct{}) (string, chan Credentials) {
 	uuid, _ := uuid.NewV4()
 
-	hostUuid := fmt.Sprintf("%s-%s", uuid.String(), host.PrivateHostname)
-	host.UUID = hostUuid
+	host.UUID = uuid.String()
 
 	if len(host.PrivateHostname) == 0 {
 		host.PrivateHostname = hostname
```

Comment:
Why not keeping that but putting it after the Privateostname check, it would be better isn't it?